src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenPage.tsx

```diff
@@ -0,0 +1,163 @@
+import Link from '@admin/components/Link';
+import { useConfig } from '@admin/contexts/ConfigContext';
+import {
+  releaseApiDataSetDetailsRoute,
+  releaseApiDataSetPreviewRoute,
+  ReleaseDataSetPreviewTokenRouteParams,
+  ReleaseDataSetRouteParams,
+} from '@admin/routes/releaseRoutes';
+import ApiDataSetPreviewTokenCreateForm from '@admin/pages/release/data/components/ApiDataSetPreviewTokenCreateForm';
+import ApiDataSetPreviewTokenDetails from '@admin/pages/release/data/components/ApiDataSetPreviewTokenDetails';
+import previewTokenQueries from '@admin/queries/previewTokenQueries';
+import apiDataSetQueries from '@admin/queries/apiDataSetQueries';
+import previewTokenService from '@admin/services/previewTokenService';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import useToggle from '@common/hooks/useToggle';
+import Button from '@common/components/Button';
+import Modal from '@common/components/Modal';
+import { useQuery } from '@tanstack/react-query';
+import { generatePath, useHistory, useParams } from 'react-router-dom';
+import React from 'react';
+
+export default function ReleaseApiDataSetPreviewTokenPage() {
+  const history = useHistory();
+  const { apiDocsUrl } = useConfig();
+
+  const [modalOpen, toggleModalOpen] = useToggle(false);
+
+  const { dataSetId, previewTokenId, releaseId, publicationId } =
+    useParams<ReleaseDataSetPreviewTokenRouteParams>();
+
+  const { data: dataSet, isLoading: isLoadingDataSet } = useQuery(
+    apiDataSetQueries.get(dataSetId),
+  );
+
+  const { data: previewToken, isLoading: isLoadingPreviewTokenId } = useQuery({
+    ...previewTokenQueries.get(previewTokenId ?? ''),
+    enabled: !!previewTokenId,
+  });
+
+  const handleCreate = async (label: string) => {
+    if (!dataSet?.draftVersion) {
+      return;
+    }
+    const token = await previewTokenService.createPreviewToken({
+      label,
+      dataSetVersionId: dataSet?.draftVersion?.id,
+    });
+
+    history.push(
+      generatePath<ReleaseDataSetPreviewTokenRouteParams>(
+        releaseApiDataSetPreviewRoute.path,
+        {
+          publicationId,
+          releaseId,
+          dataSetId,
+          previewTokenId: token.id,
+        },
+      ),
+    );
+  };
+
+  const handleRevoke = async (id: string) => {
+    await previewTokenService.revokePreviewToken(id);
+
+    history.push(
+      generatePath<ReleaseDataSetPreviewTokenRouteParams>(
+        releaseApiDataSetPreviewRoute.path,
+        {
+          publicationId,
+          releaseId,
+          dataSetId,
+        },
+      ),
+    );
```

Comment:
Think there's a bug here. I'm finding that if I create a token, then revoke it, I'll end up back on the initial preview landing page, but the 'Generate API token' modal will already be open again.

Imagine this is because the modal state hasn't updated correctly and as we're currently redirecting to the same page component.

This should be resolvable by splitting out separate pages for the initial landing page and the page for viewing the preview token (as per earlier comment).